src/labels_transfer/xgboost/README.md

```diff
@@ -0,0 +1,53 @@
+# XGBoost Labels Transfer
+
+This component transfers labels from a reference dataset to an query using a boosting method [XGBoost](https://xgboost.readthedocs.io/en/stable/).
+
+# Input
+
+- `.h5mu` files for reference and query as specified in [config](./config.vsh.yaml)
+- List of targets (labels) to predict from `.obs` slot of the reference. Common choice for HLCA would be `["ann_level_1", "ann_level_2", "ann_level_3", "ann_level_4", "ann_level_5", "ann_finest_level"]`
+- models output directory
+
+For the description of other input arguments, refer to the [config](./config.vsh.yaml).
+
+# Output
+
+`.h5mu` file with the query dataset that contains the transferred annotations
+- The annotations for each of the `targets` will be stored as new columns in the `.obs` data frame with the suffix `_pred` by default
+- The uncertainty of the prediction for each target will be stored as another column in `.obs` with the suffix `_uncertainty`
+- `labels_transfer` entry is added to the `.uns` dictionary of the output file, which contains information about the method used and its parameters. Keys of this dictionary are labels from `targets` list. Values are dictionaries with the following keys:
```

Comment:
Can you add a `--output_uns_key` to allow to choose the name of this entry?

Reply:
Sure, makes sense!